### PR TITLE
vocabbuilder.koplugin: support adding to tap gesture menu

### PR DESCRIPTION
**Pull request by weijiuqiao**
This PR makes it possible to open the plugin by taps or gestures.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9634)

### CODE REVIEW

File: plugins/vocabbuilder.koplugin/main.lua [hunk 1]
```diff
@@ -1443,12 +1445,24 @@ local VocabBuilder = WidgetContainer:extend{
...
+end
+
+function VocabBuilder:prepareWidgetToShow()
+    if self.builder_widget then
+        self.builder_widget:reloadItems()
+    else
+        -- We initiate the widget with proper
+        -- callback definition for reload_item
             local reload_items = function(builder_widget)
                 builder_widget.reload_time = os.time()
```
Comment (poire-z) [inline plugins/vocabbuilder.koplugin/main.lua:1467]: Thank you for trying to make the diff smaller :)
I had looked at some previous commit and the indentation shifts was awful for the diff).
But this indentation hiccup is noticable :)
I prefer this trick though. May be make it leaner, just double indenting the function content ?
```lua
    else
        -- We initiate the widget with proper
        -- callback definition for reload_item
        local reload_items = function(builder_widget)
                builder_widget.reload_time = os.time()
```

File: plugins/vocabbuilder.koplugin/main.lua [hunk 2]
```diff
@@ -1443,12 +1445,24 @@ local VocabBuilder = WidgetContainer:extend{
...
 function VocabBuilder:addToMainMenu(menu_items)
     menu_items.vocabbuilder = {
         text = _("Vocabulary builder"),
         callback = function()
+            self:onShowVocabBuilder()
+        end
+    }
+end
+
+function VocabBuilder:prepareWidgetToShow()
```
Comment (poire-z) [inline plugins/vocabbuilder.koplugin/main.lua:1460]: May be just :setupWidget() ? (I think we have other stuff doing it the same way).

File: plugins/vocabbuilder.koplugin/main.lua [hunk 3]
```diff
@@ -1443,12 +1445,24 @@ local VocabBuilder = WidgetContainer:extend{
...
+    if self.builder_widget then
```
Comment (poire-z) [inline plugins/vocabbuilder.koplugin/main.lua:1461]: Why "builder_" in self.builder_widget ? You could just name it self.widget or self.voca_widget.
Oh, "builder" as in VocabularyBuilder :) I thought you might have picked this from our tools/wbuilder.lua :)
Anyway, "builder" in this context of building a widget :), a reader could think it's some kind of factory and get confused.

**Review by poire-z — COMMENTED** [review]
Tested enough for it to go into v2022.10 ?

File: plugins/vocabbuilder.koplugin/main.lua (hunk 1 above)
Comment (weijiuqiao) [reply plugins/vocabbuilder.koplugin/main.lua]: Done.

File: plugins/vocabbuilder.koplugin/main.lua (hunk 2 above)
Comment (weijiuqiao) [reply plugins/vocabbuilder.koplugin/main.lua]: Done.

File: plugins/vocabbuilder.koplugin/main.lua (hunk 3 above)
Comment (weijiuqiao) [reply plugins/vocabbuilder.koplugin/main.lua]: Changed to `widget`. Right, the builder pattern, the wording can be somewhat misleading.

**Comment by weijiuqiao** [comment]
>Tested enough for it to go into v2022.10 ?

Yes. Have being using it a lot :)

File: plugins/vocabbuilder.koplugin/main.lua [hunk 4]
```diff
@@ -1498,31 +1512,38 @@ function VocabBuilder:addToMainMenu(menu_items)
...
             end
 
-            self.builder_widget = VocabularyBuilderWidget:new{
+            self.widget = VocabularyBuilderWidget:new{
                 title = _("Vocabulary builder"),
                 select_items_callback = function(obj, start_idx, end_idx)
                     DB:select_items(obj, start_idx, end_idx)
                 end,
                 reload_items_callback = reload_items
             }
```
Comment (poire-z) [inline plugins/vocabbuilder.koplugin/main.lua:1529]: Bit of cheating here with the indentation (from what I can notice with github web UI. I think these 9 lines should be shifted left.
It's small enough that all of it appearing in the diff should be ok.

Comment (weijiuqiao) [reply plugins/vocabbuilder.koplugin/main.lua]: Done

**Review by poire-z — APPROVED** [review]

**Pull request merged**